src/main/java/com/devotedmc/ExilePearl/listener/PlayerListener.java

```diff
@@ -208,6 +211,26 @@ public void onItemSpawn(EntityPortalEvent e) {
 		}
 	}
 
+	// Listen for hoppers picking up pearls
+	@EventHandler
+	public void onInventoryPickupItem(InventoryPickupItemEvent e) {
+		ExilePearl pearl = pearlApi.getPearlFromItemStack(e.getItem().getItemStack());
+		if(pearl != null){
+			if (e.getInventory().getType() == InventoryType.HOPPER) {
```

Comment:
Only thought of this after merging, but could you add an additional check here that prevents hoppers carts from picking them up?